erpnext/stock/doctype/serial_and_batch_bundle/serial_and_batch_bundle.py

```
@@ -1458,22 +1459,31 @@ def validate_batch_quantity(self):
 		if frappe.db.get_single_value("Stock Settings", "allow_negative_stock_for_batch"):
 			return
 
+		date_msg = ""
+		if posting_datetime:
+			date_msg = _(" as of {0}").format(format_datetime(posting_datetime))
+
+		msg = _(
+			"""
+			The Batch {0} of an item {1} has negative stock in the warehouse {2}{3}.
+			Please add a stock quantity of {4} to proceed with this entry.
+			If it is not possible to make an adjustment entry, please enable ‘Allow Negative Stock for Batch’ in Stock Settings to proceed. However, enabling this setting may lead to negative stock in the system, so please ensure the stock levels are adjusted as soon as possible to maintain the correct valuation rate."""
```

Ruff (0.15.0)

[warning] String contains ambiguous `‘` (LEFT SINGLE QUOTATION MARK). Did you mean ``` (GRAVE ACCENT)?

(RUF001)